### PR TITLE
isort, black, flake8, and remove circular dependencies

### PR DESCRIPTION
**Pull request by yosoyjay**
Changes motivated to test installation and enable usage of the package.

Specifically:
- Changes to imports to prevent circular dependencies.
- Logging initialized in package `__init__.py` with module loggers inheriting from that root logger.
- Apply flake8 ignoring E501.  Remaining issues with W503, E266, E722, W606, and F541.
- Apply black for formatting.
- Apply isort for formatting imports.
- Add environment.yml for package dependencies.

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  - pandas
+  - xarray
+  - requests
+  - flake8
+  - isort
+  - erddapy
+  - pytest
+  - ipython
+  - aiohttp
+  - intake-xarray
```
Comment (kthyng) [inline environment.yml:17]: Also need https://github.com/intake/intake-xarray

File: ocean_data_gateway/__init__.py [hunk 2]
```diff
@@ -1,34 +1,37 @@
...
-# import ocean_data_gateway.readers.erddap
-# import readers.axds
-# import readers.local
-#
-# from .gateway import (gateway)
-# import Data
-# from .ErddapReader import (ErddapReader, region)
-# from .axdsReader import (axdsReader)
-# from .localReader import (localReader)
+base_path = Path.home() / ".ocean_data_gateway"
```
Comment (kthyng) [inline ocean_data_gateway/__init__.py:10]: Is the slash (/) here operating system neutral? Not sure about the syntax but I was using `.joinpath` to try to work on windows.

File: ocean_data_gateway/__init__.py [hunk 3]
```diff
@@ -1,34 +1,37 @@
...
+base_path.mkdir(exist_ok=True)
+
+logs_path = base_path / "logs"
+logs_path.mkdir(exist_ok=True)
+
+catalogs_path = base_path / "catalogs"
```
Comment (kthyng) [inline ocean_data_gateway/__init__.py:16]: It's hard to look through all of these formatting changes to see — did you update all of these names in the respective files to be consistent?

**Review by kthyng — COMMENTED** [review]
Did you run the tests?

File: ocean_data_gateway/__init__.py (hunk 2 above)
Comment (yosoyjay) [reply ocean_data_gateway/__init__.py:10]: Yeah, the [`/` is actually an operator](https://docs.python.org/3/library/pathlib.html#operators), so it is OS independent.

File: ocean_data_gateway/__init__.py (hunk 3 above)
Comment (yosoyjay) [reply ocean_data_gateway/__init__.py:16]: Yeah, sorry for conflating the renaming there with what was supposed to be focused on installation and the ability to run tests.  Renaming is consistent throughout the package.

**Comment by yosoyjay** [comment]
> Did you run the tests?

Yup, that's how I found the circular imports.  Passes all the tests.

File: environment.yml (hunk 1 above)
Comment (yosoyjay) [reply environment.yml:17]: It's the last dependency listed.

Comment (kthyng) [reply environment.yml:17]: duh sorry that I missed that!

File: ocean_data_gateway/__init__.py (hunk 2 above)
Comment (kthyng) [reply ocean_data_gateway/__init__.py:10]: excellent

File: ocean_data_gateway/__init__.py (hunk 3 above)
Comment (kthyng) [reply ocean_data_gateway/__init__.py:16]: 👍

**Comment by kthyng** [comment]
Ok thanks! I'll go ahead and merge now.

**Pull request merged**